### DISCUSSION TITLE
Mass delete streamer recording files

### DISCUSSION THREAD

**Discussion by ChandigarIT** [AzuraCast/AzuraCast#7090] [Q&A]
I had a case recently where a DJ kept dropping the stream ever 30-50 seconds (we fixed the problem, it was the buffer settings in BUTT) but it has left us with around 90 files that need to be deleted. Is there a way to do this WITHOUT having to click "Delete>Delete" for each and every one of them?

Many thanks in advance.

![image](https://github.com/AzuraCast/AzuraCast/assets/108827431/7bc76632-88c9-49a1-96ab-12d230e0ea1b)

Development idea ...  A "select box" and delete button the same as in Media>Music Files would be useful,

**Comment by BusterNeece** [comment]
@ChandigarIT Right now there's no way to do this without using the API, but it's certainly a valid feature request...if you don't mind, could you submit it over at https://features.azuracast.com/ ?

**Reply by ChandigarIT** [reply]
Done, thanks.

**Comment by JeffersonStarseed** [comment]
@BusterNeece , I had this same problem with one of the Azuracast servers I set up for one of my friends.  Same thing as ChandiGarlT has descibed above and the files look just like she has displayed in the screenshot above.  What causes or triggers Azuracast to make all these short 15 to 30 second recordings? (instead of the normal triggering-on when live stream connection is detected and triggering-off when connection is disconnected)? From what ChandiGarlT describes it's a "buffer" setting issue from the encoder in whatever streaming software we're using I guess.  In the case of my friend, he is using MIXXX (not BUTT).  **_What would then be an acceptable "setting" for stream buffer, in order to prevent this from happening?_**

...and yes they are impossible to delete individually (as it is a VERY SLOW process), and it fills up the drive space on the Linux server.  Any resolution for this issue would be highly appreciated.

....But even more importantly **_for immediate resolve, what command do I use in command-line in GLISH to delete the unwanted recordings_?**  Are these recorded files able to be deleted using a command in GLISH? Trying to delete all these 15 second recordings from within Azuracasts dashboard is going to be almost an impossible or (best case scenario) a ....extremely long task.

**Reply by BusterNeece** [reply]
@JeffersonStarseed Batch deletion of broadcast recordings is now available via the web UI as of the latest Rolling Release.

**Reply by JeffersonStarseed** [reply]
Where is the new command or "button" located at in the web UI @BusterNeece ?

**Reply by BusterNeece** [reply]
@JeffersonStarseed If you're on the latest Rolling Release, there are checkboxes next to each live broadcast, with the option to delete selected at the top:

![image](https://github.com/user-attachments/assets/d730f4e0-8a3f-474e-86b5-aaa272bed9a3)

**Reply by JeffersonStarseed** [reply]
Apologies for the confusion.  The stations I administrate were not on Rolling Release but on Stable Release.  I upgraded all stations to use "Rolling Release" (as outlined in the online manual).  I had success with all stations except one.

As for the station in question for this discussion, ....I was able to go in and saw the updated controls (as pictured above) and I was able to mass-delete over 15,000 files.  Problem fixed.  Thank you for your help on this.  That was huge, and I'm sure this will come in handy for anyone else who has issues with burst-recordings building-up due to wrong buffer settings in their streaming software they're using.

However, I have one last problem.  My primary station just went caput after switching to the "Rolling Release" and updating Azuracast.  No problems though, I've already performed a cloud-host backup/restore to 4 days ago.  But do you think the problem had anything to do with that station having and using Stereo Tool?  Should I shut down Liquid Soap and uninstall Stereo Tool before re-updating that station to the latest rolling release? ...and then re-install Stereo Tool and it's preset?

**Reply by BusterNeece** [reply]
@JeffersonStarseed Without seeing logs or other info, I can't tell you what's causing that one station to go offline. We haven't changed much about Stereo Tool, though users have said switching to the new Liquidsoap module version helped with issues sometimes. If you continue to experience problems, you should create a new issue (complete with ample logs).

**Reply by JeffersonStarseed** [reply]
Thank you for your very quick response.  I just wanted to say I highly appreciate you and what you have here with Azuracast.  Once again thank you so much.

**Comment by marekdziem** [comment]
Yes a mass select / delete hetre would be very useful

**Reply by BusterNeece** [reply]
Read above; it's already in the latest Rolling Release.
